### PR TITLE
Feature/price model integration

### PR DESCRIPTION
**Pull request by gorold**
### Context
Most updated documentation: https://www.notion.so/Portal-API-ea2b289968a147b1ade07a350b9fe9b6

### Changes
- Refer to documentation

### Considerations
- Endpoints not listed in documentation are not created yet, wait for next PR.
- Note: /datablocks/<uuid:pk>/dataviz/ is not working yet.

### CODE REVIEW

File: secretsauce/apps/portal/models.py [hunk 1]
```diff
@@ -48,7 +50,7 @@ class DataBlock(models.Model):
         related_name='data_blocks'
     )
     name = models.CharField(max_length=200, unique=True)
-    upload = models.FileField(upload_to='uploads/')
+    upload = models.FileField(upload_to=obfuscate_upload_link)
```
Comment (eugenechia95) [inline secretsauce/apps/portal/models.py:54]: As mentioned, can we add a type field to this? To indicate whether it is training dataset or costdataset

File: secretsauce/apps/portal/views.py [hunk 2]
```diff
@@ -91,113 +87,74 @@ class ProjectDetail(generics.RetrieveUpdateDestroyAPIView):
...
+        if project.cost_sheet:
+            items = project.items.all()
+            for constraint_param in serializer.data:
+                try:
+                    item_id = constraint_param.get('item_id')
+                    item_name = items.get(item_id=item_id).name
+                except Item.DoesNotExist:
+                    item_name = None
+                constraint_param['item_name'] = item_name
+        return Response(serializer.data)
```
Comment (eugenechia95) [inline secretsauce/apps/portal/views.py:153]: Can explain what this section is for?

File: secretsauce/apps/portal/models.py (hunk 1 above)
Comment (gorold) [reply secretsauce/apps/portal/models.py:54]: Cost dataset should be uploaded using /portal/project/<uuid:pk>/items/, they are not the same as DataBlock. They have different headers, etc.

File: secretsauce/apps/portal/views.py (hunk 2 above)
Comment (gorold) [reply secretsauce/apps/portal/views.py:153]: Item names are obtained from FullItemRec.csv (aka cost sheet). There is only 1 cost sheet uploaded per project. This is to find the item names from the project items and match according to item_id

**Pull request merged**